src/otx/algo/segmentation/backbones/mscan.py

```diff
@@ -0,0 +1,473 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""MSCAN backbone for SegNext model."""
+
+from __future__ import annotations
+
+from pathlib import Path
+from typing import TYPE_CHECKING, Any
+
+import torch
+from mmengine.model import BaseModule
+from torch import nn
+
+from otx.algo.modules import build_activation_layer, build_norm_layer
+from otx.algo.utils.mmengine_utils import load_checkpoint_to_model, load_from_http
+
+if TYPE_CHECKING:
+    from torch import Tensor
+
+
+def drop_path(x: Tensor, drop_prob: float = 0.0, training: bool = False) -> torch.Tensor:
+    """Drop paths (Stochastic Depth) per sample.
+
+    Args:
+        x (Tensor): The input tensor.
+        drop_prob (float): Probability of the path to be zeroed. Default: 0.0
+        training (bool): The running mode. Default: False
+    """
+    if drop_prob == 0.0 or not training:
+        return x
+    keep_prob = 1 - drop_prob
+    # handle tensors with different dimensions, not just 4D tensors.
+    shape = (x.shape[0],) + (1,) * (x.ndim - 1)
+    random_tensor = keep_prob + torch.rand(shape, dtype=x.dtype, device=x.device)
+    return x.div(keep_prob) * random_tensor.floor()
+
+
+class DropPath(nn.Module):
+    """DropPath."""
+
+    def __init__(self, drop_prob: float = 0.1):
+        """Drop paths (Stochastic Depth) per sample.
+
+        Args:
+            drop_prob (float): Probability of the path to be zeroed. Default: 0.1
+        """
+        super().__init__()
+        self.drop_prob = drop_prob
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        """Forward function."""
+        return drop_path(x, self.drop_prob, self.training)
```

Comment:
Should we keep DropPath? It appears redundant alongside timm models (an existing dependency). Can we substitute it with timm.models.layers.DropPath for consistency?

Reply:
We should revisit everything what can be deleted / omitted. But maybe it is better to do after FF and basic functionality.